src/vs/base/common/processes.ts

```diff
@@ -109,6 +109,7 @@ export function sanitizeProcessEnvironment(env: IProcessEnvironment, ...preserve
 	const keysToRemove = [
 		/^ELECTRON_.+$/,
 		/^VSCODE_(?!(PORTABLE|SHELL_LOGIN|ENV_REPLACE|ENV_APPEND|ENV_PREPEND)).+$/,
+		/^VSCODE_(PWSH|ZSH|BASH)_ACTIVATE$/,
```

Comment:
Just tested/debugged this, and I dont think this does anything. 
The envs from the parameter didn't even contain these VSCODE_ACTIVATE var. 

This is weird because from env extension, we set it via env collection api at `applyProcessCreation`: https://github.com/microsoft/vscode-python-environments/blob/4a1a772112da2484927597672b7359d0d6ba8992/src/features/terminal/shells/bash/bashEnvs.ts#L21